Readme.md

```diff
@@ -87,6 +91,36 @@ test-timeout = 300
 test-no-reboot = true
 ```
 
+### Inner workings
+The `bootimage` command first reads the `CARGO_MANIFEST_DIR` environment variable to find out where the `Cargo.toml` of the current project is located.
+Then it parses the `Cargo.toml` file to read bootimage specific configuration data out of it. It then proceeds to build the current cargo project.
+Afterwards it looks if a crate named `bootloader` is defined as dependency. It reads its `Cargo.toml` and looks for more bootimage specific configuration data. These fields must be available:
+```toml
+[features]
+binary = []
+
+[package.metadata.bootloader]
+# The target specification, the default can be printend with:
+# rustc +nightly -Z unstable-options --print target-spec-json --target i686-unknown-linux-gnu
+target = "i686-unknown-linux-gnu.json"
```

Comment:
I'm not sure what you mean with the "default" here. I would probably just explain the field as "The target that the bootloader crate should be compiled for."